spigot-impl/src/main/java/me/aurium/beetle/spigot/SpigotOptions.java

```diff
@@ -0,0 +1,104 @@
+package me.aurium.beetle.spigot;
+
+import me.aurium.beetle.api.BeetleOptions;
+import me.aurium.beetle.api.task.sync.BlockingBehavior;
+import me.aurium.beetle.defaults.task.sync.TimerSleepBehavior;
+
+import java.util.Objects;
+import java.util.concurrent.Executor;
+
+public class SpigotOptions implements BeetleOptions {
```

Comment:
```suggestion
public final class SpigotOptions implements BeetleOptions {
```
To make the options in SpigotOptions truly immutable, either the getter methods or the class should be final. I suggest making the class final.